python/src/iceberg/types.py

```diff
@@ -146,7 +146,7 @@ class NestedField(IcebergType):
     name: str = field()
     field_type: IcebergType = field()
     is_optional: bool = field(default=True)
-    doc: Optional[str] = field(default=None, repr=False)
+    doc: Optional[str] = field(default=None, repr=False, compare=False, hash=False)
```

Comment:
@rdblue do we want to compare the docs as well? I know that for Avro we only compare the fields that are part of the canonical representation. I caught this in a test where my example didn't have the docs 🤔

Reply:
We compare docs in the Java implementation. That's because equality for types is that the type matches exactly, including documentation.

The question about whether a field is "equal" if the doc string is different is the reason why we don't implement `equals` in Java's `Schema` class. We could do the same here if you want.

Reply:
Let's include it here for now. I'll revert this change and update the tests 👍🏻